src/poetry/core/version/pep440/version.py

```diff
@@ -201,13 +201,11 @@ def next_postrelease(self) -> "PEP440Version":
         )
 
     def next_devrelease(self) -> "PEP440Version":
-        if self.is_prerelease():
+        if self.is_devrelease():
             dev = self.dev.next()
         else:
             dev = ReleaseTag(RELEASE_PHASE_DEV)
-        return self.__class__(
-            epoch=self.epoch, release=self.release, pre=self.pre, dev=dev
-        )
+        return self.__class__(epoch=self.epoch, release=self.release, dev=dev)
```

Comment:
```suggestion
        return self.__class__(epoch=self.epoch, release=self.release, pre=self.pre, post=self.post, dev=dev)
```

Let's retain the pre and post release for now. Handling of next pre-release, post-release when dev is set can be addressed in another PR.